Sub section to intro course (#44)
open-sauced/intro: 07-Types-of-contributions.md

Reviewer: The file title still needs to be updated. If you look at the titles, you'll notice the first word in the title is lowercase. So in this case, it should be `07-types-of-contributions.md`. 

![screenshot of titles](https://github.com/open-sauced/intro/assets/34313413/100acdd6-b549-46d9-8d9e-d663d03dbf7f)

# Types of contributions
In this chapter, we will discuss the types of contributions that open-source projects involve.

Mostly, there are two types of contributions:

- Non-coding type of contribution

- Coding type of contribution

>    ## Non-Coding Type Of Contributions

Reviewer: Remove the white space here.

### Ways to use Blogging as a Open Source Contribution

- You can write about the latest features of an open-source project, such as a new version of the software or a new plugin.

Example: [Announcing the 0.2.0 release of Analog](https://dev.to/analogjs/announcing-the-020-release-of-analog-aa1)

- Love teaching? You can write a tutorial on how to install and configure the software.

Example: [How To Make a Delicious Contributing Guide](https://dev.to/opensauced/how-to-make-a-delicious-contributing-guide-4bp3)

- A blog post could be written about the impact of an open-source project, such as how it has been used to make a difference in the world.

Example: [Retaining the best engineers](https://opensauced.pizza/blog/retaining-the-best-engineers)

- A blog post could be written on how to navigate your way around open-source as a new contributor, such as how to get started with open-source.

Example: [Open Source 101: A Beginner's Guide to Getting Started](https://opensauced.pizza/blog/open-source-101-a-beginner's-guide-to-getting-started)

> ## Coding Type Of Contributions

Reviewer: I think we should change it to Coding and Non-coding Types of Contributions

> Coding contributions are the heart of open-source development. They entail writing, modifying, or improving the actual source code of an open-source project, directly impacting the functionality and quality of the software. Contributors collaborate with project maintainers and the community to ensure that their code contributions align with project goals and coding standards. Also, open-source projects typically have guidelines and processes for accepting and integrating coding contributions, such as code reviews and continuous integration.

Reviewer: I have a hard time thinking of it as the heart. And tbh, I may have said this before, but I think we should adjust our language here.

Coding contributions power open-source development.

Example: [Adding chart](https://github.com/rupali-codes/LinksHub/pull/1632#event-10288094144)

- Updating the project's documentation to reflect new changes

Example: [Add Dev Cards to Getting Started](https://github.com/open-sauced/docs/issues/156)

- Translating the project's documentation into other languages
  
Example: [Translate the course into another language](https://github.com/open-sauced/intro/issues/16)

- Fixing typos and grammatical errors in the documentation

Example: [Adding information about possessives
](https://github.com/StyleGuides/WritingStyleGuide/pull/519)

- Improving the formatting of the documentation

Example: [Incorrect Link Destination in Documentation](https://github.com/open-sauced/docs/pull/144)
  
- Adding missing information to the documentation

Example: [Add More Information to Use This Template in the README
](https://github.com/open-sauced/100-days-of-oss-template/pull/8)

Contributing to documentation is a great way to help open-source projects and make them more accessible to everyone. If you have the skills and the interest, I encourage you to get involved!

### Graphic Design

> Graphic design is a type of open-source contribution that involves creating visual assets, such as logos, icons, illustrations, website designs, and other graphical elements, for open-source projects. These visual assets play a significant role in enhancing the overall user experience and branding of the project. Graphic design contributions are important because they can make open-source software more appealing, user-friendly, and recognizable.

Reviewer 2: ```suggestion
Love drawing or painting? Consider doing Graphic design. This open-source contribution involves creating visual assets, such as logos, icons, illustrations, website designs, and other graphical elements, for open-source projects. These visual assets play a significant role in enhancing the overall user experience and branding of the project. Graphic design contributions can make open-source software more appealing, user-friendly, and recognizable.
```

- **User Experience (UX) Design**: UX designers focus on improving the overall user experience by considering how users interact with the software. This includes designing user flows, navigation, and usability testing.

Example: [Improved the user experience of the highlights input form](https://github.com/open-sauced/app/pull/1362)

- **Accessibility Design**: Ensuring that the project's graphical elements are accessible to all users, including those with disabilities, is an important aspect of graphic design. Designers can create accessible color schemes and ensure that visual elements are perceivable and operable.

Example: [Redesign footer
](https://github.com/cesko-digital/pomahejukrajine-web/pull/222)

- **Responsive Design**: Designing visuals that work well on various devices and screen sizes is crucial for modern web applications. Responsive design ensures that the project looks good and functions properly on both desktop and mobile devices

Example: [Writing about an OSS project creates more visibility for the project and the maintainer, which can lead to better discoverability and support.](https://github.com/open-sauced/app/pull/1553)

### Blog Writing

> Blog writing as a type of open-source contribution involves creating and publishing blog posts, articles, or tutorials related to open-source software, projects, technologies, or best practices. Like documentation, you don't necessarily have to know how to code to write these posts. 

Reviewer 2: [nitpick] I gave the beginning a little bit of humor. It can help appeal to contributors with this skill.
```suggestion
If you're more of a creative writer, then blog writing is your perfect match.This type of open-source contribution involves creating and publishing blog posts, articles, or tutorials related to open-source software, projects, technologies, or best practices. Like documentation, you don't necessarily have to know how to code to write these posts. 
```

### Ways to use Blogging as a Open Source Contribution

- You can write about the latest features of an open-source project, such as a new version of the software or a new plugin.

Example: [Announcing the 0.2.0 release of Analog](https://dev.to/analogjs/announcing-the-020-release-of-analog-aa1)

- Love teaching? You can write a tutorial on how to install and configure the software.

Example: [How To Make a Delicious Contributing Guide](https://dev.to/opensauced/how-to-make-a-delicious-contributing-guide-4bp3)

- A blog post could be written about the impact of an open-source project, such as how it has been used to make a difference in the world.

Example: [Retaining the best engineers](https://opensauced.pizza/blog/retaining-the-best-engineers)

- A blog post could be written on how to navigate your way around open-source as a new contributor, such as how to get started with open-source.

Example: [Open Source 101: A Beginner's Guide to Getting Started](https://opensauced.pizza/blog/open-source-101-a-beginner's-guide-to-getting-started)

## Coding Type Of Contributions

> These contributions in the context of open-source software development involve writing, modifying, or improving the actual source code of an open-source project.
> Coding contributions are the heart of open-source development, as they directly impact the functionality and quality of the software. Contributors collaborate with project maintainers and the community to ensure that their code contributions align with project goals and coding standards. Open-source projects typically have guidelines and processes for accepting and integrating coding contributions, such as code reviews and continuous integration.

Reviewer 2: ```suggestion
Coding contributions are the heart of open-source development. They entail writing, modifying, or improving the actual source code of an open-source project,directly impacting the functionality and quality of the software. Contributors collaborate with project maintainers and the community to ensure that their code contributions align with project goals and coding standards. Also, open-source projects typically have guidelines and processes for accepting and integrating coding contributions, such as code reviews and continuous integration.
```
